src/main/java/org/opensearch/neuralsearch/processor/normalization/ScoreNormalizationTechnique.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.neuralsearch.processor.normalization;
+
+import java.util.List;
+
+import lombok.AllArgsConstructor;
+
+import org.opensearch.neuralsearch.search.CompoundTopDocs;
+
+/**
+ * Collection of techniques for score normalization
+ */
+@AllArgsConstructor
+public enum ScoreNormalizationTechnique {
+
+    /**
+     * Min-max normalization method.
+     * nscore = (score - min_score)/(max_score - min_score)
+     */
+    MIN_MAX(MinMaxScoreNormalizationMethod.getInstance());
```

Comment:
Same here as mentioned above. Could we remove the wrapping with enum?

Reply:
I think enum is needed, please check my comment